src/include/duckdb/function/cast/default_casts.hpp

```diff
@@ -83,6 +83,27 @@ struct ListBoundCastData : public BoundCastData {
 	}
 };
 
+struct StructBoundCastData : public BoundCastData {
+	StructBoundCastData(vector<BoundCastInfo> child_casts, LogicalType target_p)
+	    : child_cast_info(move(child_casts)), target(move(target_p)) {
+	}
+
+	vector<BoundCastInfo> child_cast_info;
+	LogicalType target;
+
+public:
+	unique_ptr<BoundCastData> Copy() const override {
+		vector<BoundCastInfo> copy_info;
+		for (auto &info : child_cast_info) {
+			copy_info.push_back(info.Copy());
+		}
+		return make_unique<StructBoundCastData>(move(copy_info), target);
+	}
+};
+
+unique_ptr<BoundCastData> BindStructToStructCast(BindCastInput &input, const LogicalType &source,
+                                                 const LogicalType &target); // TODO put in struct
```

Comment:
is this TODO still supposed to be here?

Reply:
ahh thanks, I'd forgotten that I wanted to put this function in the StructBoundCastData struct